tests/mock/test_referral_storage.cairo

```diff
@@ -1,145 +1,388 @@
+//! Test file for `src/mock/referral_storage.cairo`.
+
+// *********************************************************************************************
+// *                                       IMPORTS                                             *
+// *********************************************************************************************
+// Core lib imports.
+use snforge_std::{declare, start_prank, stop_prank, ContractClassTrait};
 use starknet::{ContractAddress, contract_address_const};
 
+// Local imports.
 use satoru::data::data_store::{IDataStoreDispatcher, IDataStoreDispatcherTrait};
-use satoru::role::role_store::{IRoleStoreDispatcher, IRoleStoreDispatcherTrait};
-use satoru::mock::referral_storage::{IReferralStorageDispatcher, IReferralStorageDispatcherTrait};
+use satoru::deposit::deposit::Deposit;
 use satoru::event::event_emitter::{IEventEmitterDispatcher, IEventEmitterDispatcherTrait};
+use satoru::mock::referral_storage::{IReferralStorageDispatcher, IReferralStorageDispatcherTrait};
 use satoru::mock::governable::{IGovernableDispatcher, IGovernableDispatcherTrait};
-
-use satoru::role::role;
-use satoru::deposit::deposit::Deposit;
-use satoru::tests_lib::teardown;
-use satoru::utils::span32::{Span32, Array32Trait};
+use satoru::referral::referral_tier::ReferralTier;
 use satoru::referral::referral_utils;
+use satoru::role::role_store::{IRoleStoreDispatcher, IRoleStoreDispatcherTrait};
+use satoru::tests_lib;
+
+// *********************************************************************************************
+// *                                      TEST LOGIC                                           *
+// *********************************************************************************************
+#[test]
+fn given_initialize_when_already_intialized_then_works() {
+    let (_, _, data_store, event_emitter, referral_storage, _) = setup();
+    referral_storage.initialize(event_emitter.contract_address);
+    referral_storage.initialize(event_emitter.contract_address);
+    referral_storage.initialize(event_emitter.contract_address);
```

Comment:
We should not be able to initialize more than 1 time. Let's add a check in the `initialize` function of `Governable` contract that check `gov` field in storage is set to 0 and throw an error if it's not the case

Reply:
Ah yea, wasn't sure if this was intended or not.
Updated!